lexer.go

```diff
@@ -202,8 +202,8 @@ func isWhitespace(ch byte) bool {
 }
 
 // isNewLine returns whether a character is a newline.
-func isNewLine(ch byte) bool {
-	return ch == '\n' || ch == '\r'
+func isNewLine(ch [2]byte) bool {
```

Comment:
What do you think about passing the lexer directly to this function? The same pattern is being used every time we call it:

```go
isNewLine([2]byte{l.ch, l.peekChar()})
```

The comment says `character` but it's actually not.